shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/RouteCardData.kt

```diff
@@ -306,90 +306,51 @@ data class RouteCardData(
          */
         fun addStaticStopsData(stopIds: List<String>, globalData: GlobalResponse): ListBuilder {
 
-            val routePatternsUsed = mutableSetOf<String>()
-
-            val patternsGrouped =
-                mutableMapOf<LineOrRoute, MutableMap<Stop, MutableList<RoutePattern>>>()
-
-            val parentToAllStopIds = mutableMapOf<String, MutableSet<String>>()
-
-            globalData.run {
-                stopIds.forEach { stopId ->
-                    val stop = stops[stopId] ?: return@forEach
-                    val patternsByRouteOrLine =
-                        patternsByRouteOrLine(stopId, globalData)
-                            // filter out a route if we've already seen all of its patterns
-                            .filterNot {
-                                routePatternsUsed.containsAll(
-                                    it.value.map { pattern -> pattern.id }.toSet()
-                                )
-                            }
-                    routePatternsUsed.addAll(
-                        patternsByRouteOrLine.flatMap { it.value }.map { it.id }
-                    )
-
-                    stop.parentStationId?.let { parentStationId ->
-                        parentToAllStopIds
-                            .getOrPut(parentStationId) { mutableSetOf(parentStationId) }
-                            .add(stop.id)
+            // A map of a stop to itself and any children.
+            // for standalone stops, an entry will be <standaloneStop, [standaloneStopId]>.
+            // for stations, an entry will be <station, [stationId, child1Id, child2Id, etc.]>
+            val parentToAllStops: Map<Stop, Set<String>> =
```

Comment:
pulling this up to be the first step rather than building parentToAllStopIds incrementally while iterating over all stops I think will result in some extra looping, but on a relatively small dataset, and I think worth it for the readability gains.